src/shim/kernel.rs

```diff
@@ -8,74 +8,41 @@ use fvm4::kernel::SyscallError as E4;
 use fvm_shared2::error::ErrorNumber as N2;
 use fvm_shared3::error::ErrorNumber as N3;
 use fvm_shared4::error::ErrorNumber as N4;
-use itertools::Either;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
 use std::fmt;
+use std::fmt::Debug;
 
 macro_rules! error_number {
     ($($variant:ident),* $(,)?) => {
         #[derive(Debug, Clone)]
-        #[non_exhaustive]
         pub enum ErrorNumber {
             $($variant,)*
```

Comment:
You have two different enums with duplicated variants between them - can you think of a design that gets rid of the duplication while still meeting your design goals?

Reply:
Yes.

Reply:
@aatifsyed Removed the duplicated enum. We also now preserve upstream error messages (goal 2.)